File: 3way/problem12/solution.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maxOperations = function(nums, k) {
    nums.sort((a, b) => a - b); 
    let left = 0;
    let right = nums.length - 1;
    let result = 0;
    while(left < right){
        if(nums[left] + nums[right] > k){
            right--;
        }else if(nums[left] + nums[right] < k){
            left++;
        }else{
            result++;
            left++;
            right--;
        }
    }
    return result;
};

File: 3way/problem12/Solution.java
class Solution {
    public int maxOperations(int[] nums, int k) {
        Arrays.sort(nums);
        int left = 0;
        int right = nums.length - 1;
        int result = 0;
        while(left < right){
            if(nums[left] + nums[right] > k){
                right--;
            }else if(nums[left] + nums[right] < k){
                left++;
            }else{
                result++;
                left++;
                right--;
            }
        }
        return result;
    }
}
